src/pgwire/src/protocol.rs

```diff
@@ -1076,6 +1080,27 @@ where
                     );
                     self.conn.send(msg).await?;
                 }
+
+                // Make sure any writes have been persisted as necessary.
+                if let Some(rx) = rx {
```

Comment:
I think rx should not go onto the TransactionExited struct. This whole block should be handled in sequence_end_transaction instead, which can then present the final results to the pgwire crate. sequence_end_transaction's job is to do the hard work and figure out if a transaction committed or not. I think this means that sequence_end_transaction needs to spin up a tokio task to wait for the rx, and then send back the result to pgwire, so sequence_end_transaction may also need to get passed the client `tx`.

This makes me worried about table compaction. Immediately after a table write txn is committed, coord sends a message to dataflow to advance its frontier. If persistence is happening in another task, how does it guarantee that dataflow diffs are sent to dataflow before that happens?

Reply:
I got this approach from @benesch (it's meant to mirror how tail works). What you're suggesting also makes sense to me, but I have no opinion here; happy to do whatever the two of you agree on.

Addressed the table compaction comment below.

Reply:
Okay, pushed a new commit (which I'll squash before merging) that does this the way we discussed offline. Thanks for the handholding through this!